File: NotificationSystem/NotificationSystem.Web/Controls_Update/ctrTroubleTicketReq_Update.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Data;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;
using System.Data.SqlClient;

public partial class ctrTroubleTicketReq_Update : System.Web.UI.UserControl
{

    private int m_TroubleTicketNo = 0;

    public int TroubleTicketNo
    {
        get
        {
            return m_TroubleTicketNo;
        }
        set
        {
            m_TroubleTicketNo = value;
        }
    }

    public TextBox txtTroubleTicketList { get; set; }
    protected void Page_Load(object sender, EventArgs e)
    {
        clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
        TroubleTicketReqDataTable tblTT = new TroubleTicketReqDataTable();
        try
        {
            if (Page.IsPostBack)
            {

                if (Request.Form["ctl00$MainContent$ctrTroubleTicketReq_Update$btnUpdate"] == "Update")
                    UpdateTroubleTicket();

                else

                {
                    if (m_TroubleTicketNo > 0)
                        lblResult.Text = "ID" + m_TroubleTicketNo;
                    if (lblResult.Text.Length == 0)
                        return;

                    tblTT = (TroubleTicketReqDataTable)theNotificationSystem.GetTT();
                    if (tblTT.Count == 0)
                        return;
                    {
                        var withBlock = tblTT[0];
                        withBlock.AgentID = int.Parse(this.txtAgentID.Text); withBlock.CustomerID = int.Parse(this.txtcustomerid.Text); withBlock.TroubleTicketNo = int.Parse(this.txttroubleticketno.Text);
                        cmbStatus.Text = withBlock.Status;
                        cmbType.Text = withBlock.Type;
                        withBlock.RequestDate = DateTime.Now;
                        withBlock.DueDate = DateTime.Now;
                    }
                }
            }
            this.btnUpdate.Enabled = true;
        }

        catch (Exception ex)
        {
            clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
            string NotificationBody = ex.Message + "  " + ex.StackTrace;
            SendError.SendMailMessage(NotificationBody);
            Response.Redirect("ErrorPage.aspx", false);
        }
        }
  

    public void ClearControls()
    {
        try
        {
        }


        catch (Exception)
        {
            throw;
        }
    }


    public void UpdateTroubleTicket()
    {
        try
        {
            clsTroubleTicketReq thisTroubleTicket = new clsTroubleTicketReq();

            {
                var withBlock = thisTroubleTicket;
                if (txttroubleticketno.Text.Length == 0)

                    //if (pickRequestDate.SelectedDate == null)
                    //    withBlock.RequestDate = DateTime.Now;
                    //else
                    //    withBlock.RequestDate = (DateTime)pickRequestDate.SelectedDate;
                    if (cmbType.Text == string.Empty)
                        return;

                withBlock.CustomerID = int.Parse(txtcustomerid.Text); withBlock.AgentID = int.Parse(txtAgentID.Text); withBlock.TroubleTicketNo = int.Parse(txttroubleticketno.Text);
                withBlock.Type = cmbType.Text; withBlock.Status = cmbStatus.Text;
                withBlock.DueDate = DateTime.Now; withBlock.RequestDate = DateTime.Now;


            }
            try

            {
                clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
                theNotificationSystem.UpdateTroubleTicket(thisTroubleTicket);
                lblResult.Text = "Trouble Ticket data has been updated";

            }
            catch (SqlException ex)
            {

                if (ex.Number == 2627)
                    lblResult.Text = "Ticket already exist!";
                else
                    throw new ApplicationException(ex.Message);

            }

        }
        catch (Exception ex)
        {
            clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
            string NotificationBody = ex.Message + "  " + ex.StackTrace;
            SendError.SendMailMessage(NotificationBody);
            Response.Redirect("ErrorPage.aspx", false);
        }
        PopulateControls();

    }



    private void PopulateControls()
    {
        clsNotificationSystem theNotificationSystem = new clsNotificationSystem();


        this.cmbStatus.DataSource = theNotificationSystem.GetStatusList();
        cmbStatus.DataTextField = "Status";  this.cmbStatus.DataValueField = "Status";
        this.cmbStatus.DataBind();



        this.cmbType.DataSource = theNotificationSystem.GetTypeList();
        this.cmbType.DataTextField = "Type";  this.cmbType.DataValueField = "Type";
        this.cmbType.DataBind();

    }
        protected void btnCancel_Click(object sender, EventArgs e)
        {
            Response.Redirect("TroubleTicketReq_Find.aspx", false);
        }
    }
        

File: NotificationSystem/NotificationSystem.Web/Controls_Update/ctrTroubleTicketReq_Update.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports NotificationSystem.NotificationSystem.Data.NotificationSystem
Imports NotificationSystem.NotificationSystem.Data.Classes
Imports System.Data.SqlClient
Imports Microsoft.VisualBasic

Partial Public Class ctrTroubleTicketReq_Update
    Inherits UserControl

    Private m_TroubleTicketNo As Integer = 0

    Public Property TroubleTicketNo As Integer
        Get
            Return m_TroubleTicketNo
        End Get
        Set(value As Integer)
            m_TroubleTicketNo = value
        End Set
    End Property

    Public Property txtTroubleTicketList As TextBox
    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        Dim theNotificationSystem As New clsNotificationSystem
        Dim tblTT As New TroubleTicketReqDataTable

        Try
            If Page.IsPostBack Then

                If Request.Form("ctl00$MainContent$ctrTroubleTicketReq_Update$btnUpdate") = "Update" Then
                    UpdateTroubleTicket()
                Else
                    If m_TroubleTicketNo > 0 Then lblResult.Text = "ID" & m_TroubleTicketNo.ToString()
                    If lblResult.Text.Length = 0 Then Return

                    tblTT = CType(theNotificationSystem.GetTT(), TroubleTicketReqDataTable)
                    If tblTT.Count = 0 Then Return
                    If True Then
                        Dim withBlock = tblTT(0)
                        withBlock.AgentID = Integer.Parse(txtAgentID.Text)
                        withBlock.CustomerID = Integer.Parse(txtcustomerid.Text)
                        withBlock.TroubleTicketNo = Integer.Parse(txttroubleticketno.Text)
                        cmbStatus.Text = withBlock.Status
                        cmbType.Text = withBlock.Type
                        withBlock.RequestDate = Date.Now
                        withBlock.DueDate = Date.Now
                    End If


                    btnUpdate.Enabled = True
                End If
            Else

                PopulateControls()
            End If



        Catch ex As Exception
            Dim SendError As clsNotificationSystem_Web = New clsNotificationSystem_Web()
            Dim NotificationBody = ex.Message & "  " & ex.StackTrace
            SendError.SendMailMessage(NotificationBody)
            Response.Redirect("ErrorPage.aspx", False)
        End Try
    End Sub


    Public Sub ClearControls()
        Try
            txtcustomerid.Text = String.Empty
            txtAgentID.Text = String.Empty
            txttroubleticketno.Text = String.Empty
            cmbStatus.Value = ""
            cmbType.Value = ""
            lblResult.Text = String.Empty


        Catch ex As Exception
            Throw
        End Try
    End Sub


    Public Sub UpdateTroubleTicket()

        Dim thisTroubleTicket As New clsTroubleTicketReq

        Try

            Dim withBlock = thisTroubleTicket
            If txttroubleticketno.Text.Length = 0 Then Return
            If Equals(cmbType.Text, String.Empty) Then Exit Sub

            'If (pickRequestDate.SelectedDate == null) Then
            'withBlock.RequestDate = DateTime.Now
            '  Else
            'withBlock.RequestDate = (DateTime)pickRequestDate.SelectedDate

            withBlock.CustomerID = Integer.Parse(txtcustomerid.Text)
            withBlock.AgentID = Integer.Parse(txtAgentID.Text)
            withBlock.TroubleTicketNo = Integer.Parse(txttroubleticketno.Text)
            withBlock.Type = cmbType.Text
            withBlock.Status = cmbStatus.Text
            withBlock.DueDate = Date.Now
            withBlock.RequestDate = Date.Now

            withBlock.TroubleTicketNo = Integer.Parse(Strings.Replace(txttroubleticketno.Text, "ID", ""))


            Try
                Dim theNotificationSystem As New clsNotificationSystem
                theNotificationSystem.UpdateTroubleTicket(thisTroubleTicket)
                lblResult.Text = "Trouble Ticket data has been updated"
            Catch ex As SqlException


                Throw New ApplicationException(ex.Message)


            End Try
        Catch ex As Exception
            Throw
        End Try


    End Sub



    Private Sub PopulateControls()
        Dim theNotificationSystem As clsNotificationSystem = New clsNotificationSystem()


        cmbStatus.DataSource = theNotificationSystem.GetStatusList()
        cmbStatus.DataTextField = "Status"
        cmbStatus.DataValueField = "Status"
        cmbStatus.DataBind()



        cmbType.DataSource = theNotificationSystem.GetTypeList()
        cmbType.DataTextField = "Type"
        cmbType.DataValueField = "Type"
        cmbType.DataBind()

    End Sub
    Public Sub CleanResultControl()
        lblResult.Text = String.Empty
    End Sub
    Public Sub CleanControls()

        txtcustomerid.Text = String.Empty
        txtAgentID.Text = String.Empty
        txttroubleticketno.Text = String.Empty
        cmbStatus.Value = ""
        cmbType.Value = ""
        lblResult.Text = String.Empty
    End Sub

    Protected Sub btnCancel_Click(sender As Object, e As EventArgs)
        Response.Redirect("TroubleTicketReq_Find.aspx", False)
    End Sub
End Class

